src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -2306,6 +2311,620 @@ internal unsafe void CM(B b)
+        ptr(""1"", &i);
+    }
+}", expectedOutput: "12");
+
+            verifier.VerifyIL("C.Main()", expectedIL: @"
+{
+  // Code size       24 (0x18)
+  .maxstack  3
+  .locals init (int V_0, //i
+                delegate*<string,int*,void> V_1)
+  IL_0000:  ldftn      ""void C.M(object, void*)""
+  IL_0006:  ldc.i4.2
+  IL_0007:  stloc.0
+  IL_0008:  stloc.1
+  IL_0009:  ldstr      ""1""
+  IL_000e:  ldloca.s   V_0
+  IL_0010:  conv.u
+  IL_0011:  ldloc.1
+  IL_0012:  calli      0x6
+  IL_0017:  ret
+}");
+        }
+
+        [Fact]
+        public void AddressOf_Initializer_CovariantReturns()
+        {
+            var verifier = CompileAndVerifyFunctionPointers(@"
+using System;
+public unsafe class C
+{
+    static string M1() => ""1"";
+    static int i = 2;
+    static int* M2()
+    {
+        fixed (int* i1 = &i)
+        {
+            return i1;
+        }
+    }
+
+    static void Main()
+    {
+        delegate*<object> ptr1 = &M1;
+        Console.Write(ptr1());
+        delegate*<void*> ptr2 = &M2;
+        Console.Write(*(int*)ptr2());
+    }
+}
+", expectedOutput: "12");
+
+            verifier.VerifyIL("C.Main()", expectedIL: @"
+{
+  // Code size       34 (0x22)
+  .maxstack  1
+  IL_0000:  ldftn      ""string C.M1()""
+  IL_0006:  calli      0x3
+  IL_000b:  call       ""void System.Console.Write(object)""
+  IL_0010:  ldftn      ""int* C.M2()""
+  IL_0016:  calli      0x6
+  IL_001b:  ldind.i4
+  IL_001c:  call       ""void System.Console.Write(int)""
+  IL_0021:  ret
+}");
+        }
+
+        [Fact]
+        public void AddressOf_Initializer_Overloads()
+        {
+            var verifier = CompileAndVerifyFunctionPointers(@"
+using System;
+unsafe class C
+{
+    static void M(object o) => Console.Write(""object"" + o.ToString());
+    static void M(string s) => Console.Write(""string"" + s);
+    static void M(int i) => Console.Write(""int"" + i.ToString());
+    static void Main()
+    {
+        delegate*<string, void> ptr = &M;
+        ptr(""1"");
+    }
+}", expectedOutput: "string1");
+
+            verifier.VerifyIL("C.Main()", expectedIL: @"
+{
+  // Code size       19 (0x13)
+  .maxstack  2
+  .locals init (delegate*<string,void> V_0)
+  IL_0000:  ldftn      ""void C.M(string)""
+  IL_0006:  stloc.0
+  IL_0007:  ldstr      ""1""
+  IL_000c:  ldloc.0
+  IL_000d:  calli      0x5
+  IL_0012:  ret
+}");
+        }
+
+        [Fact]
+        public void AddressOf_Initializer_Overloads_NoMostSpecific()
+        {
+            var comp = CreateCompilationWithFunctionPointers(@"
+interface I1 {}
+interface I2 {}
+static class IHelpers
+{
+    public static void M(I1 i1) {}
+    public static void M(I2 i2) {}
+}
+class C : I1, I2
+{
+    unsafe static void Main()
+    {
+        delegate*<C, void> ptr = &IHelpers.M;
+    }
+}");
+            comp.VerifyDiagnostics(
+                // (13,35): error CS0121: The call is ambiguous between the following methods or properties: 'IHelpers.M(I1)' and 'IHelpers.M(I2)'
```

Comment:
>The call is ambiguous between the following methods [](start = 42, length = 51)

We are not calling the method here. Do we report similar error for a delegate creation?

Reply:
Yep: https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQAdYwggdsgZgBoYQBLAGyIBMQBqAHwAEAmARgFgAoM3bBMFADGEAAQBJNgG8Avt179BIiS25TuozaKZsADKIAKCAPbpRU0QHMIMANyi5XR9yYFRcfFEjaWogMKiIBJsRCrmGlo6AGzaACyiALIAFJKiZACU5o5a2mwxTPHJ4r4ZogC8AHz+xvjGFBAAdADqCGTYSWQNRqbpthGarL7qXDk5OiwAPH5VUOWJfSOjmlBJuBAA7v5J6b39DnuuufrdZhbWMOVVAETFV/aOMkA==